controllers/core/openstackcontrolplane_controller.go

```diff
@@ -418,6 +441,23 @@ func (r *OpenStackControlPlaneReconciler) reconcileNormal(ctx context.Context, i
 	return ctrl.Result{}, nil
 }
 
+func (r *OpenStackControlPlaneReconciler) reconcileDelete(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, version *corev1beta1.OpenStackVersion, helper *common_helper.Helper, versionHelper *common_helper.Helper) (ctrl.Result, error) {
+	helper.GetLogger().Info("reconcile delete")
+	controllerutil.RemoveFinalizer(version, versionHelper.GetFinalizer())
+	err := versionHelper.PatchInstance(ctx, version)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	helper.GetLogger().Info(fmt.Sprintf("finalizer removed '%s' successfully", versionHelper.GetFinalizer()))
+
+	// remove instance finalizer
+	controllerutil.RemoveFinalizer(instance, helper.GetFinalizer())
```

Comment:
Same thing here with checking for return of `true`

Reply:
will do

Reply:
no need.

Reply:
here I would just let the deferred PatchInstance does its job automatically